src/game/editor/editor2.cpp

```diff
@@ -159,7 +159,7 @@ void CEditor2::UpdateAndRender()
 
 bool CEditor2::HasUnsavedData() const
 {
-	return false;
+	return m_pLastSavedHistoryEntry != m_pHistoryEntryCurrent;
```

Comment:
We can't rely on pointers here since history entries might get reused (replaced) when the stack is full. A simple flag set to 1 when a new history entry is added should suffice.